RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerMultiFit.cc

```diff
@@ -340,16 +340,23 @@ EcalUncalibRecHitWorkerMultiFit::run( const edm::Event & evt,
             << "! something wrong with EcalTimeCalibConstants in your DB? ";
 	}
 
+        int lastSampleBeforeSaturation = -1;
+        for(unsigned int iSample = 0; iSample < EcalDataFrame::MAXSAMPLES; iSample++) {
+          if ( ((EcalDataFrame)(*itdg)).sample(iSample).gainId() == 0 ) {
+            lastSampleBeforeSaturation=iSample-1;
+            break;
+          }
+        }
+
         // === amplitude computation ===
-        int leadingSample = ((EcalDataFrame)(*itdg)).lastUnsaturatedSample();
 
-        if ( leadingSample == 4 ) { // saturation on the expected max sample
+        if ( lastSampleBeforeSaturation == 4 ) { // saturation on the expected max sample
             result.emplace_back((*itdg).id(), 4095*12, 0, 0, 0);
             auto & uncalibRecHit = result.back();
             uncalibRecHit.setFlagBit( EcalUncalibratedRecHit::kSaturated );
 	    // do not propagate the default chi2 = -1 value to the calib rechit (mapped to 64), set it to 0 when saturation
             uncalibRecHit.setChi2(0);
-        } else if ( leadingSample >= 0 ) { // saturation on other samples: cannot extrapolate from the fourth one
+        } else if ( lastSampleBeforeSaturation >= 0 ) { // saturation on other samples: cannot extrapolate from the fourth one
```

Comment:
As such, if you have saturation on the first sample (lastSampleBeforeSaturation = -1) you don mark the uncalibRecHit as kSaturated and go multifit as well: is this the wanted behavior? (Easier to ask you instead of trying to understand the full logic behind...)